pkg/bot/delete.go

```diff
@@ -59,22 +52,119 @@ func (b *Bot) processDeleteCommand(update tgbotapi.Update, userID int64, r *rada
+	case DeleteMovieYes:
+		return b.handleDeleteMovieYes(update, command)
+	case DeleteMovieGoBack:
+		return b.showDeleteMovieSelection(update, command)
+	case DeleteMovieCancel:
+		b.clearState(update)
+		b.sendMessageWithEdit(command, CommandsCleared)
+		return false
+	default:
+		// Check if it starts with "TMDBID_"
+		if strings.HasPrefix(update.CallbackQuery.Data, "TMDBID_") {
+			return b.handleLDeleteMovieSelection(update, command)
+		}
+		return false
+	}
+}
+
+func (b *Bot) showDeleteMovieSelection(update tgbotapi.Update, command *userDeleteMovie) bool {
+	var keyboard tgbotapi.InlineKeyboardMarkup
+
+	// Convert the map values (movies) to a slice
+	var movies []*radarr.Movie
+	if len(command.searchResultsInLibrary) > 0 {
+		for _, movie := range command.searchResultsInLibrary {
+			movies = append(movies, movie)
+		}
+	} else {
+		for _, movie := range command.library {
+			movies = append(movies, movie)
+		}
+	}
+
+	// Sort the movies alphabetically based on their titles
+	sort.SliceStable(movies, func(i, j int) bool {
+		return utils.IgnoreArticles(strings.ToLower(movies[i].Title)) < utils.IgnoreArticles(strings.ToLower(movies[j].Title))
+	})
+
+	var movieKeyboard [][]tgbotapi.InlineKeyboardButton
+	for _, movie := range movies {
+		// Check if the movie is selected
+		isSelected := isSelectedMovie(command.selectedMovies, movie.ID)
+
+		// Create button text with or without check mark
+		buttonText := movie.Title
+		if isSelected {
+			buttonText += " \u2705"
+		}
+
+		row := []tgbotapi.InlineKeyboardButton{
+			tgbotapi.NewInlineKeyboardButtonData(buttonText, "TMDBID_"+strconv.Itoa(int(movie.TmdbID))),
+		}
+		movieKeyboard = append(movieKeyboard, row)
+	}
+
+	keyboard.InlineKeyboard = append(keyboard.InlineKeyboard, movieKeyboard...)
+
+	var keyboardConfirmCancel tgbotapi.InlineKeyboardMarkup
+	if len(command.selectedMovies) > 0 {
+		keyboardConfirmCancel = b.createKeyboard(
+			[]string{"Submit - Confirm Movies", "Cancel - clear command"},
+			[]string{DeleteMovieConfirm, DeleteMovieCancel},
+		)
+	} else {
+		keyboardConfirmCancel = b.createKeyboard(
+			[]string{"Cancel - clear command"},
+			[]string{DeleteMovieCancel},
 		)
-		b.sendMessage(editMsg)
+	}
+
+	keyboard.InlineKeyboard = append(keyboard.InlineKeyboard, keyboardConfirmCancel.InlineKeyboard...)
+
+	// Send the message containing movie details along with the keyboard
+	editMsg := tgbotapi.NewEditMessageTextAndMarkup(
+		command.chatID,
+		command.messageID,
+		"Select the movie\\(s\\) you want to delete",
+		keyboard,
+	)
+	editMsg.ParseMode = "MarkdownV2"
+	editMsg.DisableWebPagePreview = true
+	b.setDeleteMovieState(command.chatID, command)
+	b.sendMessage(editMsg)
+	return false
+}
+
+func (b *Bot) handleDeleteSearchResults(update tgbotapi.Update, searchResults []*radarr.Movie, command *userDeleteMovie) {
+	if len(searchResults) == 0 {
+		b.sendMessageWithEdit(command, "No movies found matching your search criteria")
 		return
 	}
 	if len(searchResults) > 25 {
```

Comment:
Should we make the size of the search results configurable? Or is this a telegram limitation?

Reply:
future work, requires pagination